x-pack/legacy/plugins/canvas/public/plugin.ts

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import ReactDOM from 'react-dom';
+import { Chrome } from 'ui/chrome';
+import { IModule } from 'angular';
+import { Storage } from 'ui/storage';
+import { CoreSetup, CoreStart, Plugin } from '../../../../../src/core/public';
+// @ts-ignore: Untyped Local
+import { initStateManagement, initLocationProvider } from './angular/config';
+import { CanvasRootControllerFactory } from './angular/controllers';
+// @ts-ignore: Untypled Local
+import { initStore } from './angular/services';
+// @ts-ignore: untyped local component
+import { HelpMenu } from './components/help_menu/help_menu';
+// @ts-ignore: untyped local
+import { loadExpressionTypes } from './lib/load_expression_types';
+// @ts-ignore: untyped local
+import { loadTransitions } from './lib/load_transitions';
+
+// @ts-ignore: untyped local
+import { initClipboard } from './lib/clipboard';
+
+export { CoreStart };
+/**
+ * These are the private interfaces for the services your plugin depends on.
+ * @internal
+ */
+// This interface will be built out as we require other plugins for setup
+export interface CanvasSetupDeps {} // eslint-disable-line @typescript-eslint/no-empty-interface
+export interface CanvasStartDeps {
+  __LEGACY: {
+    absoluteToParsedUrl: (url: string, basePath: string) => any;
+    formatMsg: any;
+    loadingCount: any;
+    badge: Chrome['badge'];
+    setRootController: Chrome['setRootController'];
+    storage: typeof Storage;
+    trackSubUrlForApp: Chrome['trackSubUrlForApp'];
```

Comment:
FYI we don't plan to support this API globally in the NP. You'll need to track this state yourself and boot the canvas app to the last location when it's mounted.